extensions/modules/mail/src/main/java/org/exist/xquery/modules/mail/MessageListFunctions.java

```diff
@@ -557,227 +544,166 @@ private SearchTerm[] parseChildSearchTerms( Node terms ) throws XPathException
 		}
-		
-		if( pattern != null && !pattern.isEmpty()) {
-			st = new HeaderTerm( name, pattern );
-		} else {
-			throw( new XPathException(this, "pattern attribute must be specified for term with type: " + ((Element)terms).getAttribute( "type" ) ) );
+		final String pattern = ((Element)terms).getAttribute( "pattern" );
+		if (pattern.isEmpty()) {
+			throw new XPathException(this, "pattern attribute must be specified for term with type: " + ((Element)terms).getAttribute("type"));
 		}
-		
-		return( st );
+		return new HeaderTerm( name, pattern );
 	}
 
 	private SearchTerm parseFlagTerm( Node terms ) throws XPathException
 	{
-		SearchTerm	st = null;
-		
-		String flag  = ((Element)terms).getAttribute( "flag" );
-		String value = ((Element)terms).getAttribute( "value" );
-		
-		if( StringUtils.isEmpty(value) ) {
-			throw( new XPathException(this, "value not specified for term with type: " + ((Element)terms).getAttribute( "type" ) ) );
-		}
-		
-		if( flag != null && !flag.isEmpty()) {
-			Flags flags = null;
-			
-			if( "answered".equalsIgnoreCase(flag) ) {
-				flags = new Flags( Flags.Flag.ANSWERED );
-			} else if( "deleted".equalsIgnoreCase(flag) ) {
-				flags = new Flags( Flags.Flag.DELETED );
-			} else if( "draft".equalsIgnoreCase(flag) ) {
-				flags = new Flags( Flags.Flag.DRAFT );
-			} else if( "recent".equalsIgnoreCase(flag) ) {
-				flags = new Flags( Flags.Flag.RECENT );
-			} else if( "seen".equalsIgnoreCase(flag) ) {
-				flags = new Flags( Flags.Flag.SEEN );
-			} else {
-				throw( new XPathException(this, "Invalid flag: " + flag + ", for term with type: " + ((Element)terms).getAttribute( "type" ) ) );
-			}
-				
-			st = new FlagTerm( flags, "true".equalsIgnoreCase(value) );
+		final String value = ((Element)terms).getAttribute("value");
+		if (value.isEmpty() ) {
+			throw new XPathException(this, "value not specified for term with type: " + ((Element)terms).getAttribute("type"));
+		}
+
+		final String flag  = ((Element)terms).getAttribute("flag");
+		if (flag.isEmpty()) {
+			throw new XPathException(this, "flag attribute must be specified for term with type: " + ((Element)terms).getAttribute("type"));
+		}
+		final Flags flags;
+
+		if ("answered".equalsIgnoreCase(flag) ) {
+			flags = new Flags( Flags.Flag.ANSWERED );
+		} else if( "deleted".equalsIgnoreCase(flag) ) {
+			flags = new Flags( Flags.Flag.DELETED );
+		} else if( "draft".equalsIgnoreCase(flag) ) {
+			flags = new Flags( Flags.Flag.DRAFT );
+		} else if( "recent".equalsIgnoreCase(flag) ) {
+			flags = new Flags( Flags.Flag.RECENT );
+		} else if( "seen".equalsIgnoreCase(flag) ) {
+			flags = new Flags( Flags.Flag.SEEN );
 		} else {
-			throw( new XPathException(this, "flag attribute must be specified for term with type: " + ((Element)terms).getAttribute( "type" ) ) );
+			throw new XPathException(this, "Invalid flag: " + flag + ", for term with type: " + ((Element)terms).getAttribute("type"));
 		}
```

Comment:
Here switch as function would make sense

Reply:
I propose to change this in a followup PR